added a dedicated class for converted horses

Cheaterpaul opened the pull request:
Closes #426

Cheaterpaul committed:
added a dedicated class for converted horses

maxanier commented on src/main/java/de/teamlapen/vampirism/entity/converted/EntityConvertedHorse.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public boolean isGettingSundamage(boolean forceRefresh) {
+        // TODO Auto-generated method stub
```
Guess vampire horses should also get sundamage

maxanier commented on src/main/java/de/teamlapen/vampirism/entity/converted/EntityConvertedHorse.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public boolean wantsBlood() {
+        return bloodTimer > 0;
```
But blood timer is never increased.

Also, since they will never suck blood, we probably do not need this at all

Cheaterpaul committed:
remove bloodtimer & adding sundamage

maxanier commented:
Sundamage has to be actively checked and potentially dealt in the entities update method, otherwise it does not have any affect. Vampire horses should behave similar to normal converted creatures

Cheaterpaul committed:
onLivingUpdate

maxanier commented:
Why are the horses only getting weakness while other converted animals die?
Since they might be used by players, this might make it a little easier to use them, but still it is inconsistent. Players should take care of their vampire horses. Only compromise would be giving them only very little sundamage, so players have sufficient time to bring them into safety.

maxanier commented:
Did you even test this?

a) The vampire horses don't have a vampire eye overlay anymore (No renderer is registered)
b) The entity name language key/translation is missing

It is awesome that you are contributing to the mod and doing so many PRs, but pls make sure they actually work, otherwise it would be faster if I implemented them myself (as testing is usually the part that takes the longest).

Cheaterpaul committed:
functional converted horse

Cheaterpaul committed:
client side converted horse

maxanier committed:
Fix converted horse name
Fix converted horses from being biteable after reloading the game
Stop tamed converted horses from despawning.

maxanier commented:
Entity name was still wrong
Also converted horses could be bitten after reloading the world
